guvnor-ala/guvnor-ala-docker-provider/src/test/java/org/guvnor/ala/docker/executor/DockerExecutorTest.java

```diff
@@ -0,0 +1,202 @@
+
+package org.guvnor.ala.docker.executor;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+import org.guvnor.ala.build.maven.config.impl.MavenBuildConfigImpl;
+import org.guvnor.ala.build.maven.config.impl.MavenBuildExecConfigImpl;
+import org.guvnor.ala.build.maven.config.impl.MavenProjectConfigImpl;
+import org.guvnor.ala.build.maven.executor.MavenBuildConfigExecutor;
+import org.guvnor.ala.build.maven.executor.MavenBuildExecConfigExecutor;
+import org.guvnor.ala.build.maven.executor.MavenProjectConfigExecutor;
+import org.guvnor.ala.config.BinaryConfig;
+import org.guvnor.ala.config.BuildConfig;
+import org.guvnor.ala.config.ProjectConfig;
+import org.guvnor.ala.config.ProviderConfig;
+import org.guvnor.ala.config.ProvisioningConfig;
+import org.guvnor.ala.config.RuntimeConfig;
+import org.guvnor.ala.config.SourceConfig;
+import org.guvnor.ala.docker.access.DockerAccessInterface;
+import org.guvnor.ala.docker.access.impl.DockerAccessInterfaceImpl;
+import org.guvnor.ala.docker.config.impl.ContextAwareDockerProvisioningConfig;
+import org.guvnor.ala.docker.config.impl.ContextAwareDockerRuntimeExecConfig;
+import org.guvnor.ala.docker.config.DockerProviderConfig;
+import org.guvnor.ala.docker.config.impl.DockerBuildConfigImpl;
+import org.guvnor.ala.docker.config.impl.DockerProviderConfigImpl;
+import org.guvnor.ala.docker.model.DockerRuntime;
+import org.guvnor.ala.docker.service.DockerRuntimeManager;
+import org.guvnor.ala.pipeline.Input;
+import org.guvnor.ala.pipeline.Pipeline;
+import org.guvnor.ala.pipeline.PipelineFactory;
+import org.guvnor.ala.pipeline.Stage;
+import org.guvnor.ala.pipeline.execution.PipelineExecutor;
+import org.guvnor.ala.registry.BuildRegistry;
+import org.guvnor.ala.registry.SourceRegistry;
+import org.guvnor.ala.registry.local.InMemoryBuildRegistry;
+import org.guvnor.ala.registry.local.InMemoryRuntimeRegistry;
+import org.guvnor.ala.registry.local.InMemorySourceRegistry;
+import org.guvnor.ala.runtime.Runtime;
+import org.guvnor.ala.source.git.config.impl.GitConfigImpl;
+import org.guvnor.ala.source.git.executor.GitConfigExecutor;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static java.util.Arrays.*;
+import static org.guvnor.ala.pipeline.StageUtil.*;
+import static org.junit.Assert.*;
+
+/**
+ * Simple test using the Pipeline API and the docker Provider & Executors
+ */
+public class DockerExecutorTest {
+
+    private File tempPath;
+
+    @Before
+    public void setUp() {
+        try {
+            tempPath = Files.createTempDirectory( "xxx" ).toFile();
```

Comment:
If creating the temp dir fails, wont that affect the rest of the test execution? Probably need to throw the exception and let it fail.